python/lsst/dax/obscore/obscore_exporter.py

```diff
@@ -344,15 +339,11 @@ def _make_record_batches(self, batch_size: int = 10_000) -> Iterator[RecordBatch
         if not collections:
             collections = ...
 
-        # Have to use non-public Registry interface.
-        registry = self.butler._registry  # type: ignore
-        assert isinstance(registry, SqlRegistry), "Registry must be SqlRegistry"
-        backend = SqlQueryBackend(registry._db, registry._managers, registry.dimension_record_cache)
-
-        context = backend.context()
         for dataset_type_name in self.config.dataset_types:
             _LOG.debug("Reading data for dataset %s", dataset_type_name)
-            refs = registry.queryDatasets(dataset_type_name, collections=collections, where=self.config.where)
+            refs = self.butler.registry.queryDatasets(
```

Comment:
This is rewritten in #18 to use butler.query() but rebasing should be pretty straightforward.